probe-rs/src/core/registers.rs

```diff
@@ -409,10 +437,18 @@ impl CoreRegisters {
         self.0
             .iter()
             .filter(|r| {
-                matches!(
-                    r.role,
-                    Some(RegisterRole::ArgumentAndResult(_)) | Some(RegisterRole::Argument(_))
-                )
+                if let Some(role_list) = r.roles {
+                    let mut is_argument_register = false;
+                    for role in role_list {
+                        if matches!(role, RegisterRole::Return(_) | RegisterRole::Argument(_)) {
```

Comment:
Why is a return register classified as an argument register here?

Reply:
Good catch, that was a remnant from a previous approach. Cleaned up now.